src/lib/eslint/CustomEslintEngine.ts

```diff
@@ -81,8 +81,8 @@ export class CustomEslintEngine extends AbstractRuleEngine {
 		if (rules.length > 0) {
 			this.eventCreator.createUxInfoAlwaysMessage('info.filtersIgnoredCustom', []);
 		}
-
-		const eslint = this.dependencies.createESLint(config);
+		// The config we loaded from the file should be treated as an override for whatever default configs exist.
+		const eslint = this.dependencies.createESLint({overrideConfig: config});
```

Comment:
Turns out that this structure (with `overrideConfig` etc) is actually a wrapper around the object stored in a config file. So we're manually wrapping the user's config in that structure, so that they can use a syntactically valid config file.